### PR TITLE
Issue40 circlefrom3pts

### PR DESCRIPTION
**Pull request by mattj23**
Circle3D from three points, with test code

### CODE REVIEW

File: src/Spatial/Euclidean/Circle3D.cs [hunk 1]
```diff
@@ -16,13 +16,50 @@ public Circle3D(Point3D centerPoint, UnitVector3D axis, double radius)
...
+        /// <summary>
+        /// Create a circle from three points which lie along its circumference.
+        /// </summary>
+        /// <param name="p1"></param>
+        /// <param name="p2"></param>
+        /// <param name="p3"></param>
+        public Circle3D(Point3D p1, Point3D p2, Point3D p3)
+        {
+            https://www.physicsforums.com/threads/equation-of-a-circle-through-3-points-in-3d-space.173847/
+            Vector3D p1p2 = p2 - p1;
```
Comment (JohanLarsson) [inline src/Spatial/Euclidean/Circle3D.cs]: I'd `var`here but not important.

File: src/Spatial/Euclidean/Circle3D.cs [hunk 2]
```diff
@@ -16,13 +16,50 @@ public Circle3D(Point3D centerPoint, UnitVector3D axis, double radius)
...
 
+        /// <summary>
+        /// Create a circle from three points which lie along its circumference.
+        /// </summary>
+        /// <param name="p1"></param>
+        /// <param name="p2"></param>
+        /// <param name="p3"></param>
+        public Circle3D(Point3D p1, Point3D p2, Point3D p3)
+        {
+            https://www.physicsforums.com/threads/equation-of-a-circle-through-3-points-in-3d-space.173847/
```
Comment (JohanLarsson) [inline src/Spatial/Euclidean/Circle3D.cs]: Maybe some input validation here? If they are all the same point or points on a line?

File: src/SpatialUnitTests/Euclidean/Circle3DTests.cs [hunk 3]
```diff
@@ -28,5 +28,30 @@ public void Circle2Points(string p1s, string p2s, string centers, double radius)
...
+        [TestCase("3.2522525,3.4579030,2.5739476", "2.5922429,2.5575788,4.4536494", "0.9654999,3.9766348,1.5029361", "1.4925488,3.2060788,3.1067942", 1.8557743)]
+        [TestCase("1.4907420,4.4495954,1.2951978", "2.3796522,3.6326081,0.4917802", "3.6332795,0.3779172,3.4856655", "2.3105721,2.5275616,2.8479118", 2.6033163)]
+        [TestCase("4.0537333,0.7137006,1.8633552", "0.7591453,2.7025442,4.5177595", "2.4149884,2.4450949,0.7413665", "2.3306759,1.8283553,3.0064357", 2.3490455)]
+        [TestCase("2.3488295,3.4356098,2.3024682", "2.0004679,2.0740716,4.8011107", "4.1253165,3.8784045,1.5045145", "4.3383477,2.6362129,3.7888115", 2.6089145)]
+        [TestCase("2.7730017,4.7353519,1.6764519", "4.4640538,3.8917330,1.7939499", "2.9657457,0.7421344,0.7106418", "2.8705245,2.7387444,1.1937714", 2.0564369)]
+        [TestCase("0.3309697,1.4510555,4.3876068", "3.3591227,3.2904666,1.6724443", "0.6636883,0.8606021,3.9895073", "1.7839226,2.6021881,3.1186384", 2.2464326)]
+        public void CircleFromThreePoints(string p1s, string p2s, string p3s, string centers, double radius)
```
Comment (JohanLarsson) [inline src/SpatialUnitTests/Euclidean/Circle3DTests.cs]: Nice that you write tests, much appreciated!

**Comment by JohanLarsson** [comment]
@mattj23 Looks good, I can merge as it is now or wait if you think validating the input makes sense.

**Comment by mattj23** [comment]
I'll make one more update.

**Comment by JohanLarsson** [comment]
Ok, nice, checking if they are on a line is probably tricky due to tolerances.

**Comment by mattj23** [comment]
Ok, this should be updated and ready now

**Comment by JohanLarsson** [comment]
Thank you!

**Pull request merged**